### PR TITLE
Updates the change request functions to automatically close the corresponding zendesk ticket

### PR DESCRIPTION
**Pull request by dakota002**
# Description
Adds a function to the Zendesk server file to close tickets by Id, and calls the function in both the approve and reject paths

Also fixes a typo

# Testing
- As a circular submitter, create a change request
- Check zendesk to see that the ticket was created
- As a moderator either approve or reject the request
- Go back to Zendesk and see the ticket status updated to "solved"

### CODE REVIEW

**Comment by jracusin** [comment]
Let's only automatically close zendesk tickets when accepted, not rejected, because that'll need more explanation.

File: app/lib/zendesk.server.ts [hunk 1]
```diff
@@ -43,6 +43,34 @@ export async function postZendeskRequest(request: ZendeskRequest) {
...
+        },
+      }),
+    }
+  )
+
+  if (!response.ok) {
+    console.error(response)
+    throw new Error(`Request failed with status ${response.status}`)
+  }
+  return await response.json()
```
Comment (lpsinger) [inline app/lib/zendesk.server.ts:75]: We don't really use the response, do we?

```suggestion
```

File: app/routes/circulars._archive._index/route.tsx [hunk 2]
```diff
@@ -110,22 +110,25 @@ export async function action({ request }: ActionFunctionArgs) {
...
+        },
+      })
+
       await createChangeRequest(
         {
           circularId: parseFloat(circularId),
           ...props,
           submitter,
           createdOn,
+          zendeskTicketId: zendeskResponse.request.id,
```
Comment (lpsinger) [inline app/routes/circulars._archive._index/route.tsx:128]:
```suggestion
      const {request: {id: zendeskTicketId}} = await postZendeskRequest({
        requester: { name: user.name, email: user.email },
        subject: `Change Request for Circular ${circularId}`,
        comment: {
          body: `${user.name} has requested an edit. Review at ${origin}/circulars`,
        },
      })

      await createChangeRequest(
        {
          circularId: parseFloat(circularId),
          ...props,
          submitter,
          createdOn,
          zendeskTicketId,
```

File: app/routes/circulars/circulars.server.ts [hunk 3]
```diff
@@ -539,7 +539,8 @@ export async function approveChangeRequest(
   })
 
   await deleteChangeRequestRaw(circularId, requestorSub)
-
+  if (changeRequest.zendeskTicketId)
+    await closeZendeskTicket(changeRequest.zendeskTicketId)
```
Comment (lpsinger) [inline app/routes/circulars/circulars.server.ts:543]: The only reason that we would need this is to handle change requests that pre-date the deployment of this PR, right?
```suggestion
    await closeZendeskTicket(changeRequest.zendeskTicketId)
```

Also, can we do some of these steps concurrently (`Promise.all`)?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/lib/zendesk.server.ts (hunk 1 above)
Comment (dakota002) [reply app/lib/zendesk.server.ts:75]: Nope, this one can be removed

File: app/routes/circulars/circulars.server.ts (hunk 3 above)
Comment (dakota002) [reply app/routes/circulars/circulars.server.ts:543]: Yup, and I can check on the concurrent thing. I assume probably

File: app/routes/circulars/circulars.lib.ts [hunk 4]
```diff
@@ -39,6 +39,7 @@ export interface CircularChangeRequest extends CircularMetadata {
   format: CircularFormat
   submitter: string
   createdOn: number
+  zendeskTicketId?: number
```
Comment (lpsinger) [inline app/routes/circulars/circulars.lib.ts:42]:
```suggestion
  zendeskTicketId: number
```

**Review by lpsinger — CHANGES_REQUESTED** [review]
I think you may have inadvertently added the Zendesk ticket ID to the GCN Circular.

**Comment by dakota002** [comment]
> I think you may have inadvertently added the Zendesk ticket ID to the GCN Circular.

I don't believe so, where do you see this?

**Review by lpsinger — APPROVED** [review]

**Pull request merged**